### PR TITLE
Add new alerts for alertmanager sharding mode of operation.

### PR DESCRIPTION
**Pull request by stevesg**
**What this PR does**:
Adds new alerts to cover aspects of the alertmanager pertinent to sharding operation.

**Checklist**
- [X] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

**Comment by stevesg** [comment]
Question: Shall we to add the playbooks onto this PR or open a new PR for them?

**Review by gotjosh — APPROVED** [review]
LGTM, let's get these in, for now, we can add the runbooks independently.

File: CHANGELOG.md [hunk 1]
```diff
@@ -44,6 +44,7 @@
 * [ENHANCEMENT] Added 256MB memory ballast to querier. #369
 * [ENHANCEMENT] Update gsutil command for `not healthy index found` playbook #370
 * [ENHANCEMENT] Update `etcd-operator` to latest version (see https://github.com/grafana/jsonnet-libs/pull/480). #263
+* [ENHANCEMENT] Added alertmanager alerts covering configuration syncs and sharding operation. #377
```
Comment (gotjosh) [inline CHANGELOG.md:47]:
```suggestion
* [ENHANCEMENT] Added Alertmanager alerts covering configuration syncs and sharding operation. #377
```

**Review by pracucci — APPROVED** [review]
Good job! All the alerts make sense to me 👍 

I think it's a better practice to write the playbook along with the alert instead of postponing it. You can start very easy, explaining (1) when the alert fires and (2) how to investigate (see other playbooks as an example). The "how to investigate" could even just "check out the logs to find the specific error occurred", but at least with (1) we explain when the alert fires and how the system works.

**Pull request merged**